roles/keystone/tasks/main.yml

```diff
@@ -31,9 +31,49 @@
     - etype: other
       permission: r
 
-- name: install keystone service
-  upstart_service: name=keystone user=keystone cmd=/usr/local/bin/keystone-all
-                   config_dirs=/etc/keystone
+- name: install keystone uwsgi service
+  template: src=roles/keystone/templates/etc/init/keystone.conf
```

Comment:
I did not see away to use that module to include a pre-start/post-stop for the custom upstart script.